main.tf

```diff
@@ -59,8 +59,24 @@ module "vpc" {
 
 }
 
+resource "google_compute_firewall" "private-allow-ssh" {
+  name    = "${var.deployment_name}-allow-ssh"
+  project = data.google_project.project.number
+  network = module.vpc.network_id
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22"]
+  }
+
+  source_ranges = ["0.0.0.0/0"]
```

Comment:
This would be bad from a security PoV. Any reason why we cannot use IAP?

Reply:
Because IAP, in my read of the interface, requires static instance names and lists of allowed user names or groups. Since the instance names are dynamically generated, this seems like a nonstarter.

Reply:
It requires instance names but we can get that from the [resource](https://github.com/GoogleCloudPlatform/terraform-google-load-balanced-vms/blob/f2b69d7159fac655e00fae8821c414fb96dca894/main.tf#L68)?
https://github.com/terraform-google-modules/terraform-google-bastion-host/blob/cd5a91c448348a3e83d1dba7ba9fd1342e314abe/main.tf#L106

Regarding usernames, we could allow optionally passing in a var that is an empty list by default. If a user is interested in SSH access, they can add their username.

Reply:
Right but since it is a managed instance group those names are only valid at apply time. And I can't even find the node names in the terraform state file. I don't think terraform holds on to those node names.

Reply:
Gotcha, I mistook the other instance as what we wanted to SSH into. Yeah looking at resources I don't think we can use per instance IAM here. WDYT about deploying a small bastion host if len(var.members) > 1. Users can use that as a jump host into other instances. 

The other option is granting project level tunnel accessor which is simpler but would allow access to all tunnels in a project.